src/apps/bridge/bridgePowerController.cpp

```diff
@@ -245,16 +272,33 @@ void BridgePowerController::_update(
                            "RTC is not yet set - bus off\n");
           powerBusAndSetSignal(false);
         }
+
+        // Align the first sample to UTC when the RTC finally gets set
+        checkAndUpdateRTC();
+        uint32_t currentCycleS = getEpochS();
+        if (_rtcSet && _sampleIntervalStartS > currentCycleS) {
+          time_to_sleep_ms =
+              MAX((_sampleIntervalStartS - currentCycleS) * 1000, MIN_TASK_SLEEP_MS);
+        }
```

Comment:
Prior to this PR, we would not bother aligning the first sample, so for scientific research purposes, customers would have one data point that arguably they should throw out, though they probably didn't know it. This code aligns the first sample when the bridge power controller was already configured enabled but the RTC wasn't set, and then the RTC gets set the first time. There are 2 other similar places — (1) init and (2) when the RTC was already set but power control was disabled and then gets enabled.